### PR TITLE
add gp adjust

### PR DESCRIPTION
**Pull request by lyh169**
Closes #<issue number>.

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: gasprice/kafka_proc.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+func (rp *KafkaProcessor) processor() {
+	log.Info("kafka processor start processor ")
+	defer rp.kreader.Close()
+	for {
+		select {
+		case <-rp.ctx.Done():
+			return
+		default:
```
Comment (zjg555543) [inline gasprice/kafka_proc.go:150]: How often does the kafka be queried?

File: gasprice/fixed_test.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	}
+
+	for _, tc := range testcases {
+		ctx := context.Background()
+		poolM := new(poolMock)
+		ethM := new(ethermanMock)
+		ethM.On("GetL1GasPrice", ctx).Return(tc.l1GasPrice).Twice()
+		poolM.On("SetGasPrices", ctx, tc.l2GasPrice, tc.l1GasPrice.Uint64()).Return(nil).Twice()
+		f := newFixedGasPriceSuggester(ctx, tc.cfg, poolM, ethM)
+		f.UpdateGasPriceAvg()
```
Comment (zjg555543) [inline gasprice/fixed_test.go:142]: Does it need chekck with require?

**Review by zjg555543 — APPROVED** [review]

File: gasprice/kafka_proc.go (hunk 1 above)
Comment (lyh169) [reply gasprice/kafka_proc.go:150]: yes, the inside of kafka client lib has a reconnection mechanism, and the default of query is 10s.

File: gasprice/config.go [hunk 3]
```diff
@@ -34,5 +36,15 @@ type Config struct {
 	CleanHistoryPeriod        types.Duration `mapstructure:"CleanHistoryPeriod"`
 	CleanHistoryTimeRetention types.Duration `mapstructure:"CleanHistoryTimeRetention"`
 
+	KafkaURL string `mapstructure:"KafkaURL"`
```
[the remark this answers is not in the document]
Comment (lyh169) [reply gasprice/config.go:39]: ok

File: gasprice/config.go [hunk 4]
```diff
@@ -16,6 +16,8 @@ const (
 	LastNBatchesType EstimatorType = "lastnbatches"
 	// FollowerType calculate the gas price basing on the L1 gasPrice.
 	FollowerType EstimatorType = "follower"
+	// FixedType the gas price from config that the unit is usdt
+	FixedType EstimatorType = "fixed"
```
[the remark this answers is not in the document]
Comment (lyh169) [reply gasprice/config.go:20]: I think no need restrict usdt。maybe contains usdc and others?

File: gasprice/kafka_proc.go [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+)
+
+const (
+	okbcoinId      = 7184
+	defaultTime    = 10
+	defaultMaxData = 10e6 // 10M
+)
+
+// aliyun root ca
+var rootCA = `-----BEGIN CERTIFICATE-----
```
[the remark this answers is not in the document]
Comment (lyh169) [reply gasprice/kafka_proc.go]: fixed it from file

**Pull request merged**